solutions/cf-1555B.mdx

```diff
@@ -0,0 +1,341 @@
+	int amountNeededTop = max(y2 - topPlace, 0);
+	
+	//amount needed to move bottomMost position
+	//in the rectangle to the bottomPlace
+	int amountNeededBottom = max(bottomPlace - y1, 0);
+	
+	//if the width of the two rectangles exceeds the bounds
+	//of the large rectangle, we cannot fit the two rectangles
+	//on the left or right
+	if(w1 + w2 > totalWidth)
+	{
+		amountNeededLeft = 1e9;
+		amountNeededRight = 1e9;
+	}
+	
+	//if the height of the two rectangles exceeds
+	//the height of the large rectangle, we cannot fit the
+	//the two on the top or bottom
+	if(h1 + h2 > totalHeight)
+	{
+		amountNeededTop = 1e9;
+		amountNeededBottom = 1e9;
+	}
+	
+	//neat trick to get the max out of multiple elements
+	int ans = min({amountNeededLeft, amountNeededRight, amountNeededTop, amountNeededBottom});
+	
+	//if the rectangles cannot fit in the bigger rectangles in
+	//any way, print -1
+	if(ans == 1e9)
+	{
+		cout << -1 << endl;	
+	}
+	else 
+	{
+		cout << ans << endl;	
+	}
+}
+int main(){
+	//testcases
+	int tc; cin >> tc;
+	for(int i = 0; i < tc; i++)
+	{
+		solve();
+	}
+}
+```
+
+</CPPSection>
+
+<PySection>
+
+```python
+t = int(input())
+
+for _ in range(t):
+	totalWidth, totalHeight = map(int, input().split())
+	x1, y1, x2, y2 = map(int, input().split())
+	w1 = x2 - x1
+	h1 = y2 - y1
+	w2, h2 = map(int, input().split())
+
+	ans = 1000000000;
+	#These are the optimal placements to just
+	#barely fit the second rectangle 
+
+	#minimum X value needed to fit on left
+	leftPlace = w2
+
+	#maximum Y value needed to fit on the right
+	rightPlace = totalWidth - w2
+
+	#the maximum Y value needed to fit on top
+	topPlace = totalHeight - h2
+
+	#minimum Y value needed to fit on the bottom
+	bottomPlace = h2
+
+
+	# Going through all possibilities:
+	# Top, Bottom, Left, Right
+	# Calculate the distance needed to move
+	# to the corresponding places.
+
+	# If that distance is negative, that means we don't have
+	# to move the initial rectangle
+
+
+	#amount needed to move leftMost position
+	#in the rectangle to the leftPlace
+	amountNeededLeft = max(leftPlace - x1, 0)
+
+	#amount needed to move rightMost position
+	#in the rectangle to the rightPlace
+	amountNeededRight = max(x2 - rightPlace, 0)
+
+	#amount needed to move topMost position
+	#in the rectangle to the rightPlace
+	amountNeededTop = max(y2 - topPlace, 0)
+
+	#amount needed to move bottomMost position
+	#in the rectangle to the bottomPlace
+	amountNeededBottom = max(bottomPlace - y1, 0)
+
+	# if the width of the two rectangles exceeds the bounds
+	# of the large rectangle, we cannot fit the two rectangles
+	# on the left or right
+	if(w1 + w2 > totalWidth):
+	amountNeededLeft = 1000000000
+	amountNeededRight = 1000000000
+
+	# if the height of the two rectangles exceeds
+	# the height of the large rectangle, we cannot fit the
+	# the two on the top or bottom
+	if(h1 + h2 > totalHeight):
+	amountNeededTop = 1000000000
+	amountNeededBottom = 1000000000
+
+

```

Comment:
not just here, but address the same things that i put in the cpp code

Reply:
I added spaces to the comments like you said